src/applications/disability-benefits/all-claims/tests/pages/separationTrainingPay.unit.spec.jsx

```diff
@@ -0,0 +1,128 @@
+import formConfig from '../../config/form';
+
+describe('Separation or Training Pay', () => {
+  const {
+    schema,
+    uiSchema
+  } = formConfig.chapters.veteranDetails.pages.separationTrainingPay;
+  const { defaultDefinitions: definitions } = formConfig;
+
+  it('should render', () => {
+    const form = mount(
+      <DefinitionTester
+        definitions={definitions}
+        schema={schema}
+        uiSchema={uiSchema}
+        data={{}}
+        formData={{}}/>
+    );
+
+    // Expect two questions with two radio inputs each
+    expect(form.find('.form-radio-buttons').length).to.equal(2);
+    expect(form.find('input').length).to.equal(4);
+  });
+
+  it('should fail to submit if no answers provided', () => {
+    const onSubmit = sinon.spy();
+    const form = mount(
+      <DefinitionTester
+        definitions={definitions}
+        schema={schema}
+        uiSchema={uiSchema}
+        data={{}}
+        formData={{}}
+        onSubmit={onSubmit}/>
+    );
+
+    form.find('form').simulate('submit');
+    expect(form.find('.usa-input-error-message').length).to.equal(2);
+    expect(onSubmit.called).to.be.false;
+  });
+
+  it('should submit if both questions answered with a no', () => {
+    const onSubmit = sinon.spy();
+    const form = mount(
+      <DefinitionTester
+        definitions={definitions}
+        schema={schema}
+        uiSchema={uiSchema}
+        data={{
+          'view:hasTrainingPay': false,
+          'view:hasSeparationPay': false
+        }}
+        formData={{}}
+        onSubmit={onSubmit}/>
+    );
+
+    form.find('form').simulate('submit');
+    expect(form.find('.usa-input-error-message').length).to.equal(0);
+    expect(onSubmit.calledOnce).to.be.true;
+  });
+
+  it('should not submit if both answers are Yes but no other info provided', () => {
+    const onSubmit = sinon.spy();
+    const form = mount(
+      <DefinitionTester
+        definitions={definitions}
+        schema={schema}
+        uiSchema={uiSchema}
+        data={{
+          'view:hasTrainingPay': true,
+          'view:hasSeparationPay': true
+        }}
+        formData={{}}
+        onSubmit={onSubmit}/>
+    );
+
+    form.find('form').simulate('submit');
+    expect(form.find('.usa-input-error-message').length).to.equal(2);
+    expect(onSubmit.called).to.be.false;
+  });
+
+  it('should expand more questions when answers are Yes', () => {
+    const form = mount(
+      <DefinitionTester
+        definitions={definitions}
+        schema={schema}
+        uiSchema={uiSchema}
+        data={{
+          'view:hasTrainingPay': true,
+          'view:hasSeparationPay': true
+        }}
+        formData={{}}/>
+    );
+
+    expect(form.find('.form-radio-buttons').length).to.equal(2);
+    expect(form.find('input').length).to.equal(6); // 4 radios + year input + checkbox
+    expect(form.find('select').length).to.equal(3); // month, day, service branch
+  });
+
+  it('should submit when all required info filled', () => {
```

Comment:
Is this test needed? Seems like it's mostly testing the forms library, which is tested separately.

The other tests in this file look good, though.

I think generally we want to make sure
1) All the fields we expect show up
2) We can submit with the minimum required fields
3) All the fields we expect to be required are

Reply:
I think this test is useful in that it tests that the validation logic is what we expect and there's not some hidden issue that is blocking moving to the next page.

Reply:
Yeah, I guess the thinking with this test was that clicking certain options made _more_ stuff required, so I wanted to test the base case that the form would submit with all of the conditionally required stuff filled out. But yeah, I agree, not sure there's a need for this test - the more important thing to check would be that the form didn't submit if those conditions were activated but their conditionally-required fields _weren't_ filled out. I.e., your third point. This is already tested on L67.